docs/assets/js/controllers/clipboard.ts

```diff
@@ -0,0 +1,32 @@
+(function(){
+    var application = Stimulus.Application.start();
+    application.register("clipboard", class extends Stimulus.Controller {
+        static targets = ["source"];
+        sourceTarget!: HTMLElement;
+
+        connect() {
+            super.connect();
+        };
+
+        copy() {
+            const text = this.sourceTarget.innerText;
+            navigator.clipboard.writeText(text);
+            this.handleVisible();
+        }
+
+        private handleVisible() {
+            const { scope } = this.targets;
+
+            const hideElements = scope.findAllElements('[data-hide-on-copy]');
+            const showElements = scope.findAllElements('[data-show-on-copy]');
+
+            hideElements.map(el => el.classList.add("d-none"));
+            showElements.map(el => el.classList.remove("d-none"));
+
+            setTimeout(function () {
+                hideElements.map(el => el.classList.remove("d-none"));
+                showElements.map(el => el.classList.add("d-none"));
+            }, 3000);
```

Comment:
[Is performing a mapping operation without using returned value an antipattern?](https://stackoverflow.com/questions/56903693)